iroh-willow/tests/basic.rs

```diff
@@ -0,0 +1,435 @@
+use std::time::Duration;
+
+use anyhow::Result;
+use futures_concurrency::future::TryJoin;
+use futures_lite::StreamExt;
+
+use iroh_willow::{
+    interest::{Interests, IntoAreaOfInterest},
+    proto::{
+        data_model::{Path, PathExt},
+        grouping::{Area, AreaExt},
+    },
+    session::{
+        intents::{Completion, EventKind},
+        SessionInit, SessionMode,
+    },
+};
+
+use self::util::{create_rng, insert, setup_and_delegate, spawn_two, Peer};
+
+#[tokio::test(flavor = "multi_thread")]
+async fn peer_manager_two_intents() -> Result<()> {
+    iroh_test::logging::setup_multithreaded();
+    let mut rng = create_rng("peer_manager_two_intents");
+
+    let [alfie, betty] = spawn_two(&mut rng).await?;
+    let (namespace, _alfie_user, betty_user) = setup_and_delegate(&alfie, &betty).await?;
+    let betty_node_id = betty.node_id();
+
+    insert(&betty, namespace, betty_user, &[b"foo", b"1"], "foo 1").await?;
```

Comment:
Thanks! It was a bug, and is now fixed. Didn't handle the case well where no `ReconciliationSendPayload` messages were sent after a `ReconciliationSendEntry`. Also added a test so that it doesn't break again.